Updating the node operator guide (#231)
onflow/docs: docs/building-on-flow/nodes/node-operation/index.md

Reviewer: What so you think about the following revision of line 52, which currently reads:
> An access node is staked but since it does not participate in the core Flow protocol, it does not receive any staking rewards. 

I would suggest to revise this to (edits highlighted in bold):

> An access node is **minimally staked for network security. Nevertheless, the central goal for Access Nodes is to provide functionality to its node operator. In comparison, contributing to protocol progress (e.g. routing transactions to collector clusters, relaying blocks to the unstaked peer-to-peer network, etc.) should only take up a marginal fraction an Access Node's computational resources. Furthermore, Access Node operators can freely rate-limit the amount of resources their Access Node dedicates to supporting the broader ecosystem. Therefore, Access Nodes do** not receive staking rewards.

Author: made that change

The different types of nodes are described [here](./node-roles.mdx). As node operator, you can choose to run any of the different types of node that best fits your needs.

> ## Light client a.k.a. Observer node

Author: see a preview of this page [here](https://docs-git-vishal-nodechanges-onflow.vercel.app/next/building-on-flow/nodes/node-operation)

> The light client is one of the easiest node to spin up and can be run by Dapp developers who need the latest block data available locally e.g. a wallet application that needs to track the latest block ID and height.

Reviewer 2: ```suggestion
The light client is one of the easiest nodes to spin up and can be run by Dapp developers who need the latest block data available locally, e.g. a wallet application that needs to track the latest block ID and height.
```

The observer node is not staked but still provides the same API as the access node.

To run a light client, follow this [guide](./observer-node.mdx).

## Archive node

The Archive node provides a scalable and efficient way to access the history of Flow protocol and the execution state for the current spork. Like the observer node, it too can be run by anyone without being staked or added to the approved list of nodes.
The Archive node follows the chain, stores and indexes both protocol and execution state, and allows retrieval of blocks, collections, transactions and events from the genesis of the current spork.

It also allows script execution and other read-only queries that require the execution state to be read. It can answer any queries from past data e.g. “what was the Flow account balance at height X?”, where X is several thousand blocks in the past.

The archive node is currently in beta and will be available as a GA release in H2 2023 (see [here](https://flow.com/post/flow-blockchain-node-operation-archive-node) for more)

## Full node

> A Full node is staked node that serves the network by providing data on request (i.e. gets and shares data on the p2p network).

Reviewer: Reading this, I feel there is a strong focus on full nodes sharing data. While that is true for some nodes (like ENs and in the future VNs) there are also nodes that don't talk to anyone other than staked nodes. 

I would suggest to differentiate full nodes vs light nodes, whether or not they contribute to protocol progress. I think for Collector, Consensus, Execution, and verification nodes that is pretty obvious. Nevertheless, also Access nodes contribute by routing transactions to collectors -- otherwise, the network could not ingest any transaction, which would make it practically useless. Furthermore, ANs help distribute data to the external world.

In comparison, the network could make meaningful progress without any other nodes. The terminology of light node continues to make sense in this context, because the nodes does not have to contribute any resources to network progress. In comparison, Full nodes are designed to _additionally_ provide resources for protocol-internal tasks. 

ANs are a bit of a hybrid between both worlds. Nevertheless, we indent for node operators to at least support the network with some minimal contributions, like routing transactions from other people, very rate-limited script execution, helping to distribute data to the unstaked P2P network. While an operator can run their AN solely for themselves and not provide any services to the community, it is certainly not the intended way of operating an AN. Hence, I feel the classification of an AN as full node is warranted (as you have it in your diagram).

```suggestion
In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new block. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 
```

Author: thank you - committed and reworded a lil.

> In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new block. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 

Reviewer 2: ```suggestion
In a nutshell, Full Nodes are staked network participants that drive network progress, e.g. by creating and executing new blocks. They are the primary contributors to network safety (all of them validate the correctness of the consensus process and secure the network additionally through their role-specific tasks). In comparison, Light Nodes don't contribute to the networks progress. Though, they help to secure the network by also validating the integrity of the consensus process. 
```

- The Access node is a full node that serves as an RPC node and acts as a gateway node for the network.

> - The Validator node (Collection, Consesnus, Verification and Execution) is a full node that plays a role in block generation.

Reviewer 2: ```suggestion
- The Validator node (Collection, Consensus, Verification and Execution) is a full node that plays a role in block generation.
```

Alternately, instead of running an access node, you can use the [Flow community](../access-api.mdx) access nodes or the ones run by any of the other node operators.

## Validator node

> If you want your node to participate in the nitty-gritty of Flow protocol and help in block or collection creation, transaction execution, result verification or block verification then you should run one of these four node roles -
> - Collection
> - Consensus
> - Verification
> - Execution

Reviewer: ```suggestion
You can also be a core participant in running the Flow network and contribute to securing it. Depending on your preference, you could run one or any combination of the following node roles:
- Collection Nodes collaboratively create batches of transactions (in Flow terminology collections).
- Consensus Nodes create blocks, schedule them for asynchronous execution, and commit execution results once they are verified (so called sealing). In addition, they orchestrate the Flow protocol and enforce protocol compliance. 
- Execution Nodes asynchronously execute blocks. They are the power-houses in the protocol, providing the vast computational resources available to Flow transactions. 
- Verification Nodes check the execution results in a distributed manner.
```
